### PR TITLE
Complex pawn policy

### PR DESCRIPTION
**Pull request by Excavelty**

### CODE REVIEW

File: src/core/pieces/PieceFactory.ts [hunk 1]
```diff
@@ -12,7 +14,14 @@ class PieceFactory {
...
     switch(type) {
       case PieceType.Pawn:
-        return new Pawn(color, position);
+        const pawn = new Pawn(color, position);
+        if((position.x !== 6 && color === Color.White) || (position.y !== 1 && color === Color.Black)) {
+          const movePolicies = [new MovePolicy(MoveType.Vertical, new MoveConstraints({max: 1})),
+            new MovePolicy(MoveType.Diagonal, new MoveConstraints( {max: 1} ))];
+          pawn.updatePolicies(movePolicies)
+          pawn.afterFirstMove = true;
+        }
```
Comment (Fadikk367) [inline src/core/pieces/PieceFactory.ts]: I think this logic should be inside Pawn constructor - why should PieceFactory know such implementation details of Pawn piece?

File: src/core/Board.ts [hunk 2]
```diff
@@ -45,6 +47,13 @@ class Board {
 
     this.board[to.x][to.y] = piece;
     this.board[from.x][from.y] = undefined; 
+
+    /* limit move policies to one step forward */
+    if(piece.type === PieceType.Pawn) {
+      const movePolicies = [new MovePolicy(MoveType.Vertical, new MoveConstraints({max: 2})),
```
Comment (Fadikk367) [inline src/core/Board.ts]: Probably should be `{max: 1}` here :D

**Review by Fadikk367 — CHANGES_REQUESTED** [review]

File: src/core/pieces/PieceFactory.ts [hunk 3]
```diff
@@ -12,7 +14,14 @@ class PieceFactory {
   public static createPiece(type: PieceType, color: Color, position: Position): Piece {
     switch(type) {
       case PieceType.Pawn:
-        return new Pawn(color, position);
+        const pawn = new Pawn(color, position);
+        if((position.x !== 6 && color === Color.White) || (position.y !== 1 && color === Color.Black)) {
```
Comment (Fadikk367) [inline src/core/pieces/PieceFactory.ts]: `(position.y !== 1 && color === Color.Black)` => `(position.x !== 1 && color === Color.Black)`

**Review by Fadikk367 — CHANGES_REQUESTED** [review]

File: src/core/pieces/PieceFactory.ts (hunk 1 above)
Comment (Excavelty) [reply src/core/pieces/PieceFactory.ts]: Makes sense, will put it in Pawn's constructor.

File: src/core/Board.ts (hunk 2 above)
Comment (Excavelty) [reply src/core/Board.ts]: Yes, probably, thanks for noting it. Will fix it.

**Review by Fadikk367 — APPROVED** [review]

**Pull request merged**